examples/ceed/ex1-volume.c

```diff
@@ -202,12 +202,12 @@ int main(int argc, const char *argv[]) {
   CeedOperator build_oper;
   CeedOperatorCreate(ceed, build_qfunc, CEED_QFUNCTION_NONE,
                      CEED_QFUNCTION_NONE, &build_oper);
-  CeedOperatorSetField(build_oper, "dx", mesh_restr, CEED_NOTRANSPOSE,
-                       mesh_basis,CEED_VECTOR_ACTIVE);
-  CeedOperatorSetField(build_oper, "weights", mesh_restr_i, CEED_NOTRANSPOSE,
-                       mesh_basis, CEED_VECTOR_NONE);
-  CeedOperatorSetField(build_oper, "qdata", sol_restr_i, CEED_NOTRANSPOSE,
-                       CEED_BASIS_COLLOCATED, CEED_VECTOR_ACTIVE);
+  CeedOperatorSetField(build_oper, "dx", mesh_restr, mesh_basis,
```

Comment:
So when the user sets the fields now, where is the possibility to have interlaced fields or not?

Reply:
It is set when the element restriction is created. The same elem restriction will always use the same lmode.

Reply:
Maybe I'm confused now.. But I thought that the way the restriction is applied, i.e., from an L-vec to an E-vec (operator E), or from an E-vec to an L-vec (operator E^T), could have been a totally independent concept from the choice of interlacing the fields or not.

Reply:
That is still the case. The transpose mode for E or E^T is still an option when you call `CeedElemRestrictionApply()`. Now the layout mode for interlaced fields or not will only be set once, when you call `CeedElemRestrictionCreate()`.

We found that it was very easy to create bugs by inconsistent use of the lmode field in each `CeedOperatorSetField()`.

Reply:
Ok. Thanks, sounds clear now.
Sorry, but have we ever considered calling it something else? I think the fact that both `lmode = CEED_TRANSPOSE` and `tmode = CEED_TRANSPOSE` have the same values for different concepts, can be confused, especially by new users.
Have we ever considered using something like `lmode = CEED_INTERLACED` instead of `TRANPOSE` for both concepts?

Reply:
I don't have strong feelings about that either way. @jedbrown, do you have an opinion on the clarity of using the same enum two different ways for the ElemRestrictions?

If we want to change it, I can bundle it into this PR easily.

Reply:
I think a new enum would be more clear. One cheap way is just to `#define CEED_INTERLACED CEED_TRANSPOSE`, which allows for more self-documenting code, but also doesn't prevent confusing misuse. A distinct enum would let the type system enforce that they are not interchangeable.

Reply:
I have the dentist today, but I can crank this out tomorrow morning.

Reply:
Today I Learned: GCC will not warn about using the wrong enum because different enums are still just integer constants. OSX is the only Travis env giving us these enum warnings.

Reply:
Clang has `-Wenum-conversion`, and can be used from Linux. Per this patch, it'll also be available in gcc-10. https://gcc.gnu.org/bugzilla/show_bug.cgi?id=78736#c14

Reply:
For now I'm adding Clang locally and paying extra attention to the OSX Travis builds.